File: FolioReaderKit/Carthage/Checkouts/MenuItemKit/Demo-ObjC/Demo-ObjC/ViewController.m
//
//  ViewController.m
//  Demo-ObjC
//
//  Created by CHEN Xian’an on 1/17/16.
//  Copyright © 2016 lazyapps. All rights reserved.
//

#import "ViewController.h"
@import MenuItemKit;

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UIButton *button;

@end

@implementation ViewController

- (IBAction)tapButton:(id)sender
{
  UIMenuController *controller = [UIMenuController sharedMenuController];
  __weak typeof(self) _self = self;
  UIMenuItem *textItem = [[UIMenuItem alloc] initWithTitle:@"Text" handler:^(UIMenuItem * _Nonnull item) {
    [_self showAlertWithTitle:@"Text Item tapped"];
  }];
  
  UIImage *image = [UIImage imageNamed:@"Image"];
  UIMenuItem *imageItem = [[UIMenuItem alloc]
  initWithImage:image handler:^(UIMenuItem * _Nonnull item) {
    [_self showAlertWithTitle:@"Image Item Tapped"];
  }];
  
  UIMenuItem *nextItem = [[UIMenuItem alloc] initWithTitle:@"Show More Items..." handler:^(UIMenuItem * _Nonnull item) {
    MenuItemHandler handler = ^(UIMenuItem * _Nonnull item) { [_self showAlertWithTitle:[item.title stringByAppendingString:@" Tapped"]]; };
    UIMenuItem *item1 = [[UIMenuItem alloc] initWithTitle:@"1" handler:handler];
    UIMenuItem *item2 = [[UIMenuItem alloc] initWithTitle:@"2" handler:handler];
    UIMenuItem *item3 = [[UIMenuItem alloc] initWithTitle:@"3" handler:handler];
    controller.menuItems = @[item1, item2, item3];
    [controller setMenuVisible:YES animated:YES];
  }];
  
  controller.menuItems = @[textItem, imageItem, nextItem];
  [controller setTargetRect:self.button.bounds inView:self.button];
  [controller setMenuVisible:YES animated:YES];
}

- (void)showAlertWithTitle:(NSString *)title
{
  UIAlertController *alertVC = [UIAlertController alertControllerWithTitle:title message:nil preferredStyle:UIAlertControllerStyleAlert];
  UIAlertAction *action = [UIAlertAction actionWithTitle:@"Dismiss" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {}];
  [alertVC addAction:action];
  [self presentViewController:alertVC animated:YES completion:nil];
}

@end

File: FolioReaderKit/Carthage/Checkouts/MenuItemKit/Demo/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by CHEN Xian’an on 1/16/16.
//  Copyright © 2016 lazyapps. All rights reserved.
//

import UIKit
import MenuItemKit

class ViewController: UIViewController {

  @IBOutlet var button: UIButton!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    button.addTarget(self, action: #selector(self.tapButton(_:)), forControlEvents: .TouchUpInside)
  }

  func tapButton(sender: AnyObject?) {
    let controller = UIMenuController.sharedMenuController()
    let textItem = UIMenuItem(title: "Text") { [weak self] _ in
      self?.showAlertWithTitle("text item tapped")
    }
    
    let image = UIImage(named: "Image")!
    let imageItem = UIMenuItem(image: image) { [weak self] _ in
      self?.showAlertWithTitle("image item tapped")
    }
    
    let nextItem = UIMenuItem(title: "Show More Items...") { _ in
      let handler: MenuItemHandler = { [weak self] in self?.showAlertWithTitle($0.title + " tapped") }
      let item1 = UIMenuItem(title: "1", handler: handler)
      let item2 = UIMenuItem(title: "2", handler: handler)
      let item3 = UIMenuItem(title: "3", handler: handler)
      controller.menuItems = [item1, item2, item3]
      controller.setMenuVisible(true, animated: true)
    }
    
    controller.menuItems = [textItem, imageItem, nextItem]
    controller.setTargetRect(button.bounds, inView: button)
    controller.setMenuVisible(true, animated: true)
  }

  func showAlertWithTitle(title: String) {
    let alertVC = UIAlertController(title: title, message: nil, preferredStyle: .Alert)
    alertVC.addAction(UIAlertAction(title: "Dismiss", style: .Cancel, handler: { _ in }))
    presentViewController(alertVC, animated: true, completion: nil)
  }

}
